src/packet/sym_encrypted_protected_data.rs

```diff
@@ -53,26 +52,105 @@ impl SymEncryptedProtectedData {
     }
 
     /// Encrypts the data using the given symmetric key.
-    pub fn encrypt_with_rng<R: CryptoRng + Rng>(
+    pub fn encrypt_seipdv1<R: CryptoRng + Rng>(
         rng: R,
         alg: SymmetricKeyAlgorithm,
         key: &[u8],
         plaintext: &[u8],
     ) -> Result<Self> {
-        let data = alg.encrypt_protected_with_rng(rng, key, plaintext)?;
+        let data = alg.encrypt_protected(rng, key, plaintext)?;
 
         Ok(SymEncryptedProtectedData {
             packet_version: Default::default(),
             data: Data::V1 { data },
         })
     }
 
-    /// Same as [`encrypt_with_rng`], but uses [`thread_rng`] for RNG.
-    ///
-    /// [`encrypt_with_rng`]: SymEncryptedProtectedData::encrypt_with_rng
-    /// [`thread_rng`]: rand::thread_rng
-    pub fn encrypt(alg: SymmetricKeyAlgorithm, key: &[u8], plaintext: &[u8]) -> Result<Self> {
-        Self::encrypt_with_rng(&mut thread_rng(), alg, key, plaintext)
+    /// Encrypts the data using the given symmetric key.
+    pub fn encrypt_seipdv2<R: CryptoRng + Rng>(
+        mut rng: R,
+        sym_alg: SymmetricKeyAlgorithm,
+        aead: AeadAlgorithm,
+        chunk_size: u8,
+        session_key: &[u8],
+        plaintext: &[u8],
+    ) -> Result<Self> {
+        // TODO: DRY decrypt/encrypt code duplication?
```

Comment:
what about this?

Reply:
I've DRY'd this in aa47208034f1d42cc80e87ec6d99ca2b1c153845.

(Not sure about the name and return-type of the internal helper function, though :sweat:)